lib/ansible/modules/cloud/smartos/vmadm.py

```diff
@@ -0,0 +1,600 @@
+        ip: dhcp
+        primary: true
+    internal_metadata:
+      root_pw: 'secret'
+    quota: 1
+
+- name: Delete a zone
+  vmadm:
+    alias: test_zone
+    state: deleted
+
+- name: Stop all zones
+  vmadm:
+    uuid: '*'
+    state: stopped
+'''
+
+RETURN = '''
+uuid:
+    description: UUID of the managed VM.
+    returned: always
+    type: string
+    sample: 'b217ab0b-cf57-efd8-cd85-958d0b80be33'
+alias:
+    description: Alias of the managed VM.
+    returned: When addressing a VM by alias.
+    type: string
+    sample: 'dns-zone'
+state:
+    description: State of the target, after execution.
+    returned: success
+    type: string
+    sample: 'running'
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import os
+import re
+import tempfile
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+# While vmadm(1M) supports a -E option to return any errors in JSON, the
+# generated JSON does not play well with the JSON parsers of Python.
+# The returned message contains '\n' as part of the stacktrace,
+# which breaks the parsers.
+
+
+def get_vm_prop(module, uuid, prop):
+    """Lookup a property for the given VM.
+    Returns the property, or None if not found."""
+    cmd = 'vmadm lookup -j -o {0} uuid={1}'.format(prop, uuid)
+
+    (rc, stdout, stderr) = module.run_command(cmd)
+
+    if rc != 0:
+        module.fail_json(msg=stderr)
+
+    try:
+        return json.loads(stdout)[0][prop]
+    except:
+        return None
+
+
+def get_vm_uuid(module, alias):
+    """Lookup the uuid that goes with the given alias.
+    Returns the uuid or '' if not found."""
+    cmd = 'vmadm lookup -j -o uuid alias={0}'.format(alias)
+
+    (rc, stdout, stderr) = module.run_command(cmd)
+
+    if rc != 0:
+        module.fail_json(msg=stderr)
+
+    try:
+        return json.loads(stdout)[0]['uuid']
+    except:
+        return ''
+
+
+def get_all_vm_uuids(module):
+    """Retrieve the UUIDs for all VMs"""
+    cmd = 'vmadm lookup -j -o uuid'
+
+    (rc, stdout, stderr) = module.run_command(cmd)
+
+    if rc != 0:
+        module.fail_json(msg=stderr)
+
+    try:
+        output = json.loads(stdout)
+        return list(map(lambda v: v['uuid'], output))
+    except:
+        return []
+
+
+def new_vm(module, uuid, vm_state):
+    payload_file = create_payload(module, uuid)
+
+    (rc, stdout, stderr) = vmadm_create_vm(module, payload_file)
+
+    if rc != 0:
+        changed = False
+        module.fail_json(msg='Could not create VM: {0}'.format(stderr))
+    else:
+        changed = True
+        # 'vmadm create' returns all output to stderr...
+        match = re.match('Successfully created VM (.*)', stderr)
+        if match:
+            vm_uuid = match.groups()[0]
+            if not is_valid_uuid(vm_uuid):
+                module.fail_json(msg='Invalid UUID for VM {0}?'.format(vm_uuid))
+        else:
+            module.fail_json(msg='Could not retrieve UUID of newly created(?) VM')
```

Comment:
I don't recall actually running into it, however a tool that prints it's normal output to `stderr` can be expected to do anything. So I'd rather err on the side of caution and handle such a case.